addons/website/static/src/scss/website.scss

```diff
@@ -361,6 +361,11 @@ $-seen-urls: ();
 .o_navbar_mobile {
     height: 100dvh;
     background-color: o-color('menu-custom') or o-color('menu');
+
+    // Make offcanvas close btn above mega menu
+    .btn-close {
+        z-index: 1002;
```

Comment:
About this approach, I see in `website.scss` that we sometimes use variable definition above the selector styling the component.

Note that defining the z-index in `secondary_variables.scss` will require me to expose `$zindex-offcanvas` since it's not defined yet at this state. 
1) Clean way would require me to define a `$o-zindex-offcanvas` in primary_variables + binding it in `bootstrap_overriden` it feels a bit overkill since it seems used only for these mega menus for now.

2) Compromising by setting the following in secondary_variables:
```scss 
$o-mega-menu-nav-height: 3rem !default;
$o-mega-menu-zindex: 1045 !default;
$o-mega-menu-btn-close-zindex: $o-mega-menu-zindex + 1 !default;
```
3) Keeping them in `website.scss` where we can make use of `$zindex-offcanvas` which make sense to me if we don't plan to use these variables elsewhere than this file.